### PR TITLE
[Ready For Merge] December Eris Merge

### PR DESCRIPTION
**Pull request by APWill**
## About The Pull Request
This takes us within a week of Eris development. Yay.

## Changelog
**Additions**
-Added From Serbia With Love Vendor, stocking a variety of fashionable(?) military surplus crates.
-These crates contain some previously implemented caps, but also new uniforms and armor.
-New armor variants, and some new sprites for previously existing armor and other clothing items.
-New NeoTheologist belt variants.
-Modified HUD. Status and intent is now more compact, with new elements for NSA, bionics, and sanity, some only showing when appropriate. Action buttons now have text on them instead of pictograms.
-Plastic zipties, a restraint capable of cuffing people wearing hardsuits - previously only available with the much weaker and makeshift cable ties. Also added to Sec vendors.
-Added matter guns that use material stacks for ammo.
-Added cleansing carbine that shoots cleaning foam darts. Uses biomatter for ammo.
-Added mental imprinter, a 5 TC uplink item that allows a 5 stat boost at a cost of 30 sanity.
-Added industrial grinder, that processes items like regular grinder but instead of a detachable beaker it has internal buffer - from which bottles can be created like in chemmaster. Bottles are also created automatically when there is enough of a single reagent to fill one.
-Added fake handcuffs, a prank toy favored by clowns...
-Added laser attack functionality. When using a charge weapon, hold the mouse button down to begin charging, and release to fire the weapon. You can release the charge when hovering over a target and it will fire upon them.
**New Tool Interactions**
-Multitools can be used to seal gas canisters to prevent them from being opened. Takes a decent amount of time to do. Useful for transporting canisters through public areas.
-Multitools can toggle the R&D consoles access requirement. Takes a decent amount of time, depending on the COG stat. 
-Multitools can be used to toggle the access cover lock of turrets. When open, a screwdriver can be used to access the ID slot from which you can add it to the target exceptions list. A multitool can be used to purge the target exceptions, and wirecutters can be used to attempt to disable target exceptions for even security personnel and heads of staff. Failing this can be VERY dangerous, however.
-Saws and wrenches can now be used to dismantle a locker, not just welders.
**Sanity (Yes, Still Disabled)**
-Added the new insight system. Insight is gained from any change in sanity, and when full generates a desire such as consuming food. When fulfilled, random stats are increased for a total of 15 points and stats from an oddity are added if the player possesses one (a choice is given if multiple are owned.)
-Made oddity stats randomised on spawn, making them more mysterious and unpredictable.
-Added a lengthy cooldown to obsession-related messages. No more spam.
-Removed PTSD code due to unpopularity and the performance impact.
**Mapping**
-Redesign of the chemistry lab, allowing more space within which to work freely.
**Balance/Tweaks - General**
-Given the few vending machines with generic names (i.e. 'coffee machine') proper brand names.
-Added leather gloves to the two forms of hydroponics locker.
-Made R&D console now display adjusted material costs when protolathe/imprinter are upgraded.
-Reduced the time needed to break out of fuzzy handcuffs and legcuffs. They are only toys after all.
-NSA gain and reduction is no longer instant. It is changed smoothly over a minute instead.
-NSA effects now depend on how much over limit you are. Having NSA over 200 knocks you out.
-Disabled mechs spawning with maintenance mode enabled.
-Added multitools to the technomancer and exultant toolbelts.
-Lockers can now be rigged with electropacks in the same manner as crates.
-Some structures such as  lockers, computer frames and girders can now be recycled in the smelter.
-The grinder now uses NanoUI instead of a radial menu and plain HTML combo.
-Buffed the strength of most alcohol.
-Screwed up RIGs are now easier to repair by cutting and mending all of the wires.
-Hypospray now spawns with 30u of tricordrazine.
-Backpack oxygen boxes now spawn with an injector. 
-Storage space of a box increased by 1 (one tiny item size).
-Oxygen tank storage now dispenses the tanks that were put into it manually first.
-Various improvements and sprite changes for the suit storage unit.
-Chameleon gun got insane values of illegal tech for some bizarre reason, removed now.
-Some AI module also got nerfed to more sane values.
-There will be much less money found in junk piles now, from 500 to 1000, instead of 1000 to 5000.
-Wallets now can hold discs and syringes.
-Comm relays now work on the shuttles.
-Mining drills consume less fuel now.
-Hulk now has a large fuel tank for the drills.
-Ore scanner is a scanner subtype now - so uses power cells, fits in pockets, and prints ore data.
-The junk beacon now uses its own loot table.
-Design file contracts no longer accept copy protected design files.
**Balance/Tweaks - PCs/PDAs**
-PDA light can now be toggled from the top bar - no need to go into the settings.
-New icons for some PC parts.
-Ship networking is a bit faster now.
-Boxes of basic data disks can now be bought in Print-o-Mat. It's 100 credits for a box of 7 disks, but the disks inside are just 16GQ.
-New CPU tier: advanced CPU. It's the middle ground between normal CPU and photonic CPU.
-New hacking app: "Bootkit Installer". Run it on any PC to give it access to secret software repository. You know it worked if the PC reboots.
-OldPDA carts no longer spawn in lockers.
**Balance/Tweaks - Combat & Mobs**
-Further rebalancing and refactoring of armor and bullet damage values.
-A lot of weapons were removed from material based system and put into tools, making them moddable by that. This also includes the arm blades, as well as all of the knifes, swords. mini hoe and broken bottle, for meme purposes.
-Reduced chat spam from roaches.
-Set only one stack of roaches to now spawn in lockers to improve performance.
-Changes to certain weapon names in light of new vendors.
-The Jester storyteller is now more aggressive with burrows.
-Fuhrer and Panzer roaches can no longer be slipped up with soap bars.
-Hunter roaches move slightly faster than the rest now.
-Nanite roaches may now survive in the vacuum of space.
-Nanite locust swarms are now non-dense.
-Toxic roaches no longer receive skin burns from toxins (plasma) in the air.
-Cuts blood loss values down slightly.
-Shrapnel only causes damage when not on walk intent.
-You can now work out who is a changeling using -that- test from The Thing. ; )
**Balance/Tweaks - NeoTheology**
-Buffed certain NeoTheology rituals and added a missing vigilance litany.
-Deprivation now requires the target to first be dead before uninstalling the cruciform.
**Balance/Tweaks - Antagonists**
-Completed contracts now display in round-end report.
-Feedback messages for spying sensors, mind fryers, completed contracts, and available TCs.
-New traitor items now have proper descriptions, materials and research levels.
-The uplink descriptions now clue you in on how to use the contract items.
-Some contracts pick their targets better now.
-Contracts now do in depth search on BSDM contents.
-Mind fryers should register mental breaks a lot more reliably now.
-Mind imprinters now work better if your target stat is <= 15.
**Balance/Tweaks - Voidsuits**
-Voidsuit helmets now spawn inside the voidsuit.
-Voidsuits that spawn on the map with magboots/jetpacks now spawn with those inside the voidsuit.
-You can no longer remove or add a helmet to a voidsuit.
-You no longer need to detach the helmet/tank/magboots from a voidsuit to clean it, cleaning the voidsuit itself will automatically clean its parts.
-Removes the excelsior helmet from the excelsior disk as it will now come with the armor.
-Fixes the export price for void suits, makes the new prices account for the helmet in the suit, and gives prices for emergency, syndi, and captain's soft suits.
-Adds an action button(the ui at the top left of the screen) for toggling the voidsuit helmet.
**Fixes**
-Fixed stashes sometimes being completely empty for no particular reason.
-Fixed certain items sold by Frozen Star Guns&Ammo not having a price. The bargains are over!
-Fixed one tank bombs being impossible to assemble.
-Fixed a click cooldown error related to being buckled.
-Fixed protolathe and imprinter deconstruction dropping autolathe circuit boards.
-Fixed protolathe being unable to use its chemicals.
-Fixed certain roundstart designs not unlocking properly.
-Fixed some R&D design names not displaying correctly.
-Fixed materials being wrongly lost or gained in construction/deconstruction of walls and structures.
-Fixed a range of errors relating to MPCs and MPC software.
-Fixed issues with MIRC and improved the UI substantially.
-Various grammatical fixes and enhancements.
-Fixed the grinder being non-deconstructable, and made out of a computer rather than machine frame.
-Fixes hivemind wires being unable to take over seeded tiles - also adds invasiveness as an (unobtainable) trait for plant for future use.
**Technical/Code**
-Optimized tank dispenser code.
-RIG autodoc no longer generates GC faults.
-"is_associative" helper should no longer produce runtimes.
-509 compatibility code is removed, the rest of the codebase is not 509 compatible anyway.
-JSON reader and writer are removed - twice. A single use is replaced with BYOND's JSON builtin.
-Removed useless TARGETED_LAYER define.
-Fixed ammo casings being on the wrong plane.
-Removed a lot of useless AI code.
-Changed depreciated mentions of 'lentext' to 'length'.
-Removal of a redefenition in dna2.dm.
-Solved _defines folder splitting.
-Moves more cursor code to mob level: rather than using 6 static sprites for recoil, the icon is generated procedurally. These icons are stored in an icon buffer for later use.
-Changed New in chem grenades to Initialize.
[Note: some of this might not work our end until we set up the Discord link serverside.]

### CODE REVIEW

**Comment by APWill** [comment]
A note that I am not adding the new 'insight' mechanic (https://github.com/discordia-space/CEV-Eris/pull/4350) as I believe this will add to the sanity-related annoyances rather than decrease them.

**Comment by APWill** [comment]
https://github.com/discordia-space/CEV-Eris/pull/4358/files

This hud stuff is interesting but due to the focus on sanity-related stuff within it I'm not touching it for now.

**Comment by APWill** [comment]
https://github.com/discordia-space/CEV-Eris/pull/4366
Hmm. Does our server not like Travis?

File: code/modules/clothing/head/helmet.dm [hunk 1]
```diff
@@ -82,11 +89,14 @@
 	name = "handmade combat helmet"
 	desc = "It looks like it was made from a bucket and some steel. Uncomfortable and heavy but better than nothing."
 	icon_state = "hm_helmet"
-	armor = list(melee = 35, bullet = 30, energy = 30, bomb = 0, bio = 0, rad = 0)
-	flags_inv = HIDEEARS
```
Comment (skraaak) [inline code/modules/clothing/head/helmet.dm]: Do note these changes remove the HIDEEARS flags on helmets and none of the new helmets have that flag either.

File: code/modules/webhook/webhook.dm [hunk 2]
```diff
@@ -1,15 +1,82 @@
...
+		var/query_string = "type=lobbyalert"
+		query_string += "&key=[url_encode(config.webhook_key)]"
+		query_string += "&msg=[url_encode(message)]"
+		query_string += "&color=[url_encode(color)]"
+		if(sender)
+			query_string += "&from=[url_encode(sender)]"
+		world.Export("[config.webhook_url]?[query_string]")
+
+ADMIN_VERB_ADD(/client/proc/discord_msg, R_ADMIN, TRUE)
+/client/proc/discord_msg()
```
Comment (skraaak) [inline code/modules/webhook/webhook.dm]: Let's just be sure this isn't going to ping the _Eris_ Discord...

**Comment by skraaak** [comment]
Aside those two comments of concern, everything looks in order, as much of an absolute slog it was to read. Do note that that this nerfs most of the armors.

File: code/modules/clothing/head/helmet.dm (hunk 1 above)
Comment (APWill) [reply code/modules/clothing/head/helmet.dm]: /obj/item/clothing/head/helmet is already set to HIDEEARS so it isn't required to have that on /obj/item/clothing/head/helmet/handmade. ; )

**Comment by APWill** [comment]
Webhook commit reverted for now until we can figure this out. Ready for merge.

**Comment by king5327** [comment]
Two weeks*

**Pull request merged**